### DISCUSSION TITLE
How to get list of all connected devices?

### DISCUSSION THREAD

**Discussion by atauf** [PhilipsHue/flutter_reactive_ble#324] [Q&A]
I'd like to know if there is built in functionality to view a list (or something similar) of all connected devices. Is it in the API? I haven't been able to find it. Or, do you have to just create your own list and build your own functionality?

I have no issue getting info on a single connected device, it's just when it's multiple devices that I'm unsure.

Thanks!

**Comment by remonh87** [comment]
Unfortunately  iOS does not support this functionality as far as I know (@werediver  you are even more knowledgeable on this can you confirm?  ) so that is why we decided to now add this functionality. Btw BLE caching on the OS is quite good so it is no problem to send a connect command even though there is already a connection. 

What do you mean with being unsure about multiple devices? We designed this library for the specific purpose to execute operations on multiple peripherals if this does not work for you log an issue so we can check it out.

**Reply by atauf** [reply]
@remonh87 I'm just saying that I'm unsure as to how to get a list of all the connected devices. Is that something I should be able to do using the package, or do I just need to keep track myself?

NOTE: I have connected to multiple devices successfully, so I'm not having any issue with that.

**Reply by chandlery54** [reply]
I have this same question. I am working on a solution to just keep track of devices I connected to myself, but it seems weird to have to check if devices I have connected to are still available in the discovered devices list. It would be nice if there was a way to see a list of devices that already have a connection so we can display them separately from other discovered devices.
